src/AutoRest.CSharp/Common/Output/Builders/JsonSerializationMethodsBuilder.cs

```diff
@@ -113,7 +113,7 @@ private static MethodBodyStatement WriteProperty(Utf8JsonWriterExpression utf8Js
                 utf8JsonWriter.WritePropertyName(serialization.SerializedName),
                 serialization.CustomSerializationMethodName is {} serializationMethodName
                     ? InvokeCustomSerializationMethod(serializationMethodName, utf8JsonWriter)
-                    : SerializeExpression(utf8JsonWriter, serialization.ValueSerialization, serialization.Value)
+                    : SerializeExpression(utf8JsonWriter, serialization.ValueSerialization, serialization.EnumerableValue) // EnumerableValue is equal to Value if its not an array
```

Comment:
This is a potential bug prone. I would rather change `SerializeExpression` itself, specifically `JsonArraySerialization array` branch:
```cs
JsonArraySerialization array => SerializeArray(utf8JsonWriter, array, GetEnumerableExpression(array.ImplementationType, expression))
```
where
```
private static GetEnumerableExpression(CSharpType arrayImplementationType, ValueExpression expression)
{
    if (expression is TypedValueExpression typed && TypeFactory.IsReadOnlyMemory(typed))
    {
        expression = expression.NullableStructValue(typed.Type).Property(nameof(ReadOnlyMemory<byte>.Span));
    }
    return new EnumerableExpression(TypeFactory.GetElementType(arrayImplementationType), expression))
}

Reply:
In consolidation branch, `SerializeExpression` uses `TypedValueExpression`, so I'll remove the casting during merge.